README.md

```diff
@@ -1,27 +1,38 @@
-# epm
-
-Ethereum Package Managment
+# EPM
 
 [![Join the chat at https://gitter.im/ethpm/Lobby](https://badges.gitter.im/ethpm/Lobby.svg)](https://gitter.im/ethpm/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-# Goals
-
-The goal of this repository is to create a specification that can be used for
-package management for the Ethereum Virtual Machine (EVM).
+Ethereum Packaging Specification
 
-This explicitely includes:
 
-* Specification for a *manifest* file.
+## Use Cases
 
+The following use cases were used to guide the specification.
 
-This explicitely does not include:
+1. Code with zero deployed instances.
+    1. Reusable base contracts.
+2. Code with one or more addresses.
+    1. Libraries: Installation of the package involves having an importable interface for the library and linking against the proper address during deployment.
+    2. Contracts: Installation of the package involves having an importable interface and some manner of templating in the proper addresses into the local source code prior to compilation as well as other non-contract code that may interact with the installed contract.
 
-* ????
 
+It is worth pointing out that the *Library* and *Contract* use cases are functionally the same.  Once [Solidity issue #242](https://github.com/ethereum/solidity/issues/242) has been addressed this distinction may no longer need to exist.
 
-# Resources
 
-The following resources are encouraged reading before joining the discussion.
+## Open Questions
 
-* https://github.com/purescript/purescript/issues/631
-    * Long discussion on package management from the purescript community.
+* Version
+    * Should the version live in the package manifest, release manifest, or both.
+        * Case for Package
+            * Defines the *current* version of the package.
+        * Case for Release
+            * Each release should have a clearly defined version.
+        * Case for Both
+            * Seems to work well for release workflow. Package manager would
+              populate the release manifest version field from the package
+              manifest version when creating a new release.
+* Validation of the concept of Package Dependencies vs Release Dependencies
+    * When a dependency is defined at the package level it *can* be either pinned to a specific release, or possibly a range of releases such as `^0.4.0` to allow any version in the `0.4.x` line.
```

Comment:
The thing is, apparently, we can create a blockchain URI with IPFS. Now granted, perhaps this is relying too much on IPFS, but it's a fairly solid protocol. There may also be room for a combination approach...But I'm not sure how we want to go about this...As for Swarm + IPFS, there is a difference in the headers, but I believe that could be added accordingly if they are there, we just need a way to get that one hash and then have the headers added subsequently.